docs/openapi-issuer.yaml

```diff
@@ -0,0 +1,570 @@
+        id:
+          $ref: "#/components/schemas/Id"
+      required:
+        - id
+
+    RequestSignatureBody:
+      type: object
+      properties:
+        dossier_id:
+          $ref: "#/components/schemas/Id"
+        signer_id:
+          allOf:
+            - $ref: "#/components/schemas/Id"
+            - example: "01BX5ZZKBKACTAV9WEVGEMMVS0"
+        expires_at:
+          $ref: "#/components/schemas/Timestamp"
+      required:
+        - dossier_id
+        - signer_id
+
+    Document:
+      type: object
+      properties:
+        id:
+          $ref: "#/components/schemas/Id"
+        metadata:
+          $ref: "#/components/schemas/DocumentMetadata"
+        created_at:
+          $ref: "#/components/schemas/Timestamp"
+        updated_at:
+          $ref: "#/components/schemas/Timestamp"
+      required:
+        - id
+        - status
+        - metadata
+        - created_at
+        - updated_at
+
+    DocumentToBeUploaded:
+      allOf:
+        - $ref: "#/components/schemas/Document"
+        - type: object
+          properties:
+            status:
+              type: string
+              enum:
+                - WAIT_FOR_UPLOAD
+          required:
+            - status
+
+    DocumentToBeValidated:
+      allOf:
+        - $ref: "#/components/schemas/Document"
+        - type: object
+          properties:
+            status:
+              type: string
+              enum:
+                - WAIT_FOR_VALIDATION
+            uploaded_at:
+              $ref: "#/components/schemas/Timestamp"
+          required:
+            - status
+            - uploaded_at
+
+    DocumentUploaded:
+      allOf:
+        - $ref: "#/components/schemas/Document"
+        - type: object
+          properties:
+            status:
+              type: string
+              enum:
+                - READY
+            uploaded_at:
+              $ref: "#/components/schemas/Timestamp"
+            url:
+              type: string
+              format: uri
+          required:
+            - status
+            - uploaded_at
+            - url
+
+    DocumentRejected:
+      allOf:
+        - $ref: "#/components/schemas/Document"
+        - type: object
+          properties:
+            status:
+              type: string
+              enum:
+                - REJECTED
+            uploaded_at:
+              $ref: "#/components/schemas/Timestamp"
+            rejected_at:
+              $ref: "#/components/schemas/Timestamp"
+            reject_reason:
+              type: string
+          required:
+            - status
+            - uploaded_at
+            - rejected_at
+            - reject_reason
+
+    SignatureRequest:
+      type: object
+      properties:
+        id:
+          $ref: "#/components/schemas/Id"
+        dossier_id:
+          $ref: "#/components/schemas/Id"
+        expires_at:
+          $ref: "#/components/schemas/Timestamp"
+        documents:
+          type: array
+          items:
+            oneOf:
```

Comment:
```suggestion
            anyOf:
```

Reply:
why `anyOf`?

Reply:
`documents` can match one or more subschemas. There can be DocumentToBeUploaded and also DocumentUploaded in the same array

Reply:
yes, but that is the purpose of `oneOf` (if I understood correctly)

for each element it checks if the shape matches with one of the scheme provided

`anyOf` checks the shape of an object between one and more schemes

Reply:
Mhh... yes you are right!

Reply:
anyOf is not supported by the codegen